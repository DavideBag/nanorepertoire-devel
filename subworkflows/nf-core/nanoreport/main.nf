// This subworkflows is used to create the report of nanorepertoire analysis
// modules to include in this subworkflow

include { MULTIQC as MULTIQC1   } from '../../../modules/nf-core/multiqc/main.nf'
include { MULTIQC as MULTIQC2   } from '../../../modules/nf-core/multiqc/main.nf'
include { REPORT    }             from '../../../modules/local/report/main.nf'

workflow NANOREPORT {

    // with take we define the input channels

    take:
    fastqc_zip
    cutadapt_log
    report
    loop_tree
    clustersummaries
    cdr3histograms
    cdr3tables
    metadata
    options


    main:

    ch_versions = Channel.empty()

    MULTIQC1(fastqc_zip,   [], [],[], [], [])
    ch_versions = ch_versions.mix(MULTIQC1.out.versions.first())

    MULTIQC2(cutadapt_log, [], [],[], [], [])
    ch_versions = ch_versions.mix(MULTIQC2.out.versions.first())

    REPORT(report, loop_tree, clustersummaries, cdr3histograms, cdr3tables, metadata, options)



    emit:
    // TODO nf-core: edit emitted channels
    multiqc_report     = MULTIQC1.out.report            // channel: [ val(meta), [ bam ] ]
    multiqc_report1    = MULTIQC2.out.report            // channel: [ val(meta), [ bam ] ]
    analysis_report    = REPORT.out.report              // channel: [ val(meta), [ bai ] ]

    versions           = ch_versions                    // channel: [ versions.yml ]
}

