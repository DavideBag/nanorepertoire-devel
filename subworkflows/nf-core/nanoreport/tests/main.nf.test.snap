{
    "sarscov2 - bam - single_end": {
        "content": [
            {
                "0": [
                    "multiqc_report.html:md5,e9142b105b7c5e3d7e72bff7173a1de9"
                ],
                "1": [
                    "multiqc_report.html:md5,121da822f7a1e8f8c32b2b5bd1f7cf5d"
                ],
                "2": [
                    "analysis_report.html:md5,51c9876f67701196e5ff572b75f43ecd"
                ],
                "3": [
                    "versions.yml:md5,ba27257c5db36b7e5a1bd764e6788914",
                    "versions.yml:md5,f7f62760b72c39ace865313374e5c526"
                ],
                "analysis_report": [
                    "analysis_report.html:md5,51c9876f67701196e5ff572b75f43ecd"
                ],
                "multiqc_report": [
                    "multiqc_report.html:md5,e9142b105b7c5e3d7e72bff7173a1de9"
                ],
                "multiqc_report1": [
                    "multiqc_report.html:md5,121da822f7a1e8f8c32b2b5bd1f7cf5d"
                ],
                "versions": [
                    "versions.yml:md5,ba27257c5db36b7e5a1bd764e6788914",
                    "versions.yml:md5,f7f62760b72c39ace865313374e5c526"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-09T15:45:53.518645"
    }
}